### ISSUE TITLE
Need Feature request for Custom code redeem android

### ISSUE THREAD

**Issue by harisahmedDCS** [dooboolab-community/react-native-iap#2708]
**Description**
according to android you can redeem your custom code inApp  there is no function available to redeem code in app
i successfully redeem one time code through deeplink https://play.google.com/redeem?code=promo_code  but we can not redeem  custom promo code outside the app 
android reference: Custom codes can be redeemed only from within your app, while one-time codes can be redeemed through both your app and the Play store.

**Comment by arnold-drita-abm** [comment]
Yup, I just came here to ask exactly the same thing. Is there any suggested practice on how we can implement promo codes in android builds?

**Comment by YevriiMykhailo** [comment]
Approve this one. According to google documentation: https://developer.android.com/google/play/billing/promo

> For subscription promo codes, note the following:
> 
> - Custom codes can be redeemed only from within your app, while one-time codes can be redeemed through both your app and the Play store.

Also I have found next one: https://github.com/hyochan/react-native-iap/issues/1107#issuecomment-827806724

As per mentioned comment it should be possible to redeem a Custom code in app if we choose Payment method => Redeem code. But I can approve that on testing env and production (when app downloaded from app store), no such button appear (Redeem Code)

**Comment by MarwaKhaledMohamed** [comment]
I do have the same exact issue , I created a promo, but in both the testing and production environments , the 'Redeem Code' option doesn't appear.

> Approve this one. According to google documentation: https://developer.android.com/google/play/billing/promo
> 
> > For subscription promo codes, note the following:
> > 
> > * Custom codes can be redeemed only from within your app, while one-time codes can be redeemed through both your app and the Play store.
> 
> Also I have found next one: [#1107 (comment)](https://github.com/hyochan/react-native-iap/issues/1107#issuecomment-827806724)
> 
> As per mentioned comment it should be possible to redeem a Custom code in app if we choose Payment method => Redeem code. But I can approve that on testing env and production (when app downloaded from app store), no such button appear (Redeem Code)

**Comment by MarwaKhaledMohamed** [comment]
I do have the same exact issue , I created a promo, but in both the testing and production environments , the 'Redeem Code' option doesn't appear.

**Comment by MarwaKhaledMohamed** [comment]
I have the exact same issue. I created a promo, but in both the testing and production environments, the 'Redeem Code' option doesn't appear.
